core/arxiv/submission/process/checks/tex_produced/__init__.py

```diff
@@ -0,0 +1,354 @@
+
+The tex-produced check current works on both streams and files for PDFs and just files
+for Postscript files. The original code only worked with files. In NG we pass around streams
+of file content which would require writing out in order to user the file-based check.
+
+We now convert the non-seekable stream to an in-memory object and use the PyPDF2 package
+to extract info and fonts information from a PDF.
+
+There are then a specific set of information we look for in the PDF info and certain
+fonts that indicate the content was generated by some form of TeX engine.
+
+"""
+import io
+import re
+import subprocess
+import os
+import tempfile
+
+from typing import IO, List
+from arxiv.base import logging  # type: ignore
+
+from PyPDF2 import PdfFileReader, utils
+
+logger = logging.getLogger(__name__)
+
+verbose = 0
+
+
+def get_filtered_pdf_info_from_stream(stream: IO[bytes]) -> List[str]:
+    """
+    Returns select set of values from pdfinfo output for specified file.
+
+    Parameters
+    ----------
+    stream: IO[bytes
+        The stream to run pdfinfo on.
+
+    Returns
+    -------
+    List containing one entry for each selected line from pdfinfo output.
+
+    """
+    info_list = []
+
+    try:
+        pdf = PdfFileReader(stream)
+        information = pdf.getDocumentInfo()
+
+        if information.creator is not None:
+            info_list.append(bytes('Creator:        ' + information.creator, 'utf-8'))
+
+        if information.title is not None:
+            info_list.append(bytes('Title:          ' + information.title, 'utf-8'))
+
+        if information.producer is not None:
+            info_list.append(bytes('Producer:       ' + information.producer, 'utf-8'))
+        return info_list
+    except utils.PdfReadError as ex:
+        logger.error(f"get_filtered_pdf_info_from_stream: Can't read PDF: Error: {ex}")
+        raise ex
+    except Exception as ex:
+        logger.error(f"Error: Unknown {ex}")
+        raise ex
+
+    return info_list
+
+
+def get_filtered_pdf_info_from_file(filepath: str) -> List[str]:
+    """
+    Returns select set of values from pdfinfo output for specified file.
+
+    Parameters
+    ----------
+    filepath : str
+        The file to run pdfinfo on.
+
+    Returns
+    -------
+    List containing one entry for each selected line from pdfinfo output.
+
+    """
+    info_list = []
+
+    info = subprocess.run(["pdfinfo", f"{filepath}"], stderr=subprocess.PIPE,
+                          stdout=subprocess.PIPE)
+
+    lines = info.stdout.splitlines()
+
+    # Filter for only those elements we are checking
+    for line in lines:
+        if re.search(rb'^(Creator:|Producer:|Title)', line):
+            info_list.append(line)
+
+    return info_list
+
+
+# This find-font code [walk() and parts of get_pdf_fonts_from_stream()] is
+# taken from the web and originally written by Tim Arnold.
+# I have modified it for our needs. DLF2
+
+def walk(obj, fnt, emb):
```

Comment:
please add typing for the input params and return value if possible

Reply:
I will run the code checks again. I forgot about the code I borrowed.

Reply:
Cleaned up typing.